BadHabbit/BadHabbit/Kit/DesignKit/GradientButton.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  GradientButton.swift
+//  BadHabbit
+//
+//  Created by Halina Kurylchykava on 5.09.23.
+//
+
+import UIKit
+
+class GradientButton: UIButton {
+
+    override class var layerClass: AnyClass {
+        return CAGradientLayer.self
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        configureGradientLayer()
+        layer.masksToBounds = true
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        configureGradientLayer()
+    }
+
+    func configureGradientLayer() {
```

Comment:
Do we need to call this function outside? Maybe call after button initiated ?

Reply:
configureGradient was moved after initiating gradientButton

Reply:
Why do we need to call this function outside of a GradientButton? Is there any case when you use GradientButton and don't call configureGradientLayer? If no we need to make it private call configure in GradientButton

Reply:
done